x-pack/plugins/alerting/server/task_runner/create_execution_handler.ts

```diff
@@ -107,29 +109,56 @@ export function createExecutionHandler<
         }),
       }));
 
-    alertExecutionStore.numberOfGeneratedActions += actions.length;
+    alertExecutionStore.incrementNumberOfGeneratedActions(actions.length);
 
     const ruleLabel = `${ruleType.id}:${ruleId}: '${ruleName}'`;
 
     const actionsClient = await actionsPlugin.getActionsClientWithRequest(request);
     let ephemeralActionsToSchedule = maxEphemeralActionsPerRule;
 
     for (const action of actions) {
-      if (alertExecutionStore.numberOfTriggeredActions >= ruleType.config!.run.actions.max) {
-        alertExecutionStore.triggeredActionsStatus = ActionsCompletion.PARTIAL;
+      const { actionTypeId } = action;
+
+      alertExecutionStore.incrementNumberOfGeneratedActionsByConnectorType(actionTypeId);
+
+      if (alertExecutionStore.hasReachedTheExecutableActionsLimit(actionsConfigMap)) {
+        alertExecutionStore.setTriggeredActionsStatusByConnectorType({
+          actionTypeId,
+          status: ActionsCompletion.PARTIAL,
+        });
+        logger.debug(
+          `The maximum number of actions (${actionsConfigMap.default.max}) for this rule type (ruleId:${ruleId}) has been reached`
```

Comment:
Perhaps we could format this to be more consistent with existing debug messages? Something like:

```
Rule "${ruleId}" skipped scheduling action "${action.id}" because the maximum number of allowed actions has been reached.
```